### PR TITLE
Update balancing implementation

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: database/database.go [hunk 1]
```diff
@@ -916,6 +886,8 @@ func (db *Database) createInstancesTable() (err error) {
 	_, err = db.sql.Exec(`CREATE TABLE IF NOT EXISTS instances (serviceId TEXT,
                                                                 subjectId TEXT,
                                                                 instance INTEGER,
+																nodeID TEXT,
```
Comment (mlohvynenko) [inline database/database.go:889]: please fix aligment

**Review by mlohvynenko — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

File: unitstatushandler/softwaremanager.go [hunk 2]
```diff
@@ -823,11 +828,72 @@ func (manager *softwareManager) update(ctx context.Context) {
...
 	}
 
-	manager.runCond.Wait()
+	manager.runStatusCV.Wait()
+
+	if len(manager.revertServices) != 0 {
+		manager.doRevertServices()
+		manager.revertServices = nil
+
+		if err := manager.runInstances(); err != nil && updateErr == nil {
```
Comment (mykola-kobets-epam) [inline unitstatushandler/softwaremanager.go:841]: Before starting service it should be installed, I guess. But just reverting the service should be enough as the first step for recovering instance run.

Rerunning the instances can require more difficult strategy to implement.

**Review by mykola-kobets-epam — APPROVED** [review]

File: storagestate/storagestate.go [hunk 3]
```diff
@@ -484,24 +504,25 @@ func (storageState *StorageState) setupStateWatching(instanceID, stateFilePath s
...
+		params.InstanceIdent, instanceID, stateFilePath, params.StateQuota); err != nil {
 		return aoserrors.Wrap(err)
 	}
 
 	return nil
 }
 
 func (storageState *StorageState) startStateWatching(
-	instanceID string, stateFilePath string, params SetupParams,
+	instanceIdent aostypes.InstanceIdent, instanceID string, stateFilePath string, quota uint64,
```
[the remark this answers is not in the document]
Comment (al1img) [reply storagestate/storagestate.go:516]: Looks like instance ID is redundant and not used.

File: unitstatushandler/softwaremanager.go (hunk 2 above)
Comment (al1img) [reply unitstatushandler/softwaremanager.go:841]: For example?

**Pull request merged**